### DISCUSSION TITLE
Boolean tag parsing

### DISCUSSION THREAD

**Discussion by EricCraigen** [getsentry/sentry-java#2467] [General]
When setting the value of a tag to any of the following: 

"False"
"FALSE"
"FalSe"
"True"
"TRUE"
"TrUe"

The tags are not presented in the UI as a boolean value, yet all of these values could successfully be parsed into a boolean value. 

![image](https://user-images.githubusercontent.com/72918809/211932472-29bf0b99-434e-416a-8d53-64a83b4c2556.png)

I have a few questions surrounding this:

1. Is this intended? 
2. If so, why so specific that only "true" or "false" will successfully be parsed to a boolean type to be presented in the UI differently?
3. What is the purpose of treating boolean tags differently? (colored tag background) No other tags are treated this way.

**Comment by adinauer** [comment]
Hello @EricCraigen here's the code that affects the rendering:

https://github.com/getsentry/sentry/blob/82de1f15c881f6b35643be5a9109d94f348fdb0a/static/app/components/pill.tsx#L23-L47

and 

https://github.com/getsentry/sentry/blob/82de1f15c881f6b35643be5a9109d94f348fdb0a/static/app/components/pill.tsx#L74-L104

So these should only be visual aids

**Reply by ctadlock** [reply]
That logic seems flawed. Since when is `negative` an error to warrant highlighting in red? Especially since the CSS logic is almost the same for `negative` and `error`.

**Reply by adinauer** [reply]
I've created an issue and transferred it to https://github.com/getsentry/sentry/issues/43299 as this is related to the Sentry UI and not the SDK. Your feedback should be handled there.
